From 7c0d3922bb196d5ab3e89a3640e356c7cffe1939 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 23 Jan 2024 18:46:16 +0100
Subject: [PATCH] arm64: dts: st: Add EXTI1 power domains

Add the power domain RET_PD to EXTI1 to manage the support of devices
interruptions for STM32MP25 low power modes
see the "Table 94. Functionalities depending on system operating mode"
in reference manual.

As the D2 domain must be powered to have EXTI1 functional, the
LPLV-Stop1 low power modes is not supported for CpuIdle (with PSCI OSI)
when EXTI1 is used.

Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I8398e02eee4ce8c062a583e1eb7add7466aa9d16
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/393893
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp231.dtsi | 1 +
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 1 +
 2 files changed, 2 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp231.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp231.dtsi
@@ -2271,6 +2271,7 @@
 			compatible = "st,stm32mp1-exti";
 			interrupt-controller;
 			#interrupt-cells = <2>;
+			power-domains = <&RET_PD>;
 			reg = <0x44220000 0x400>;
 			interrupts-extended =
 				<&intc GIC_SPI 268 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_0 */
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2598,6 +2598,7 @@
 			compatible = "st,stm32mp1-exti";
 			interrupt-controller;
 			#interrupt-cells = <2>;
+			power-domains = <&RET_PD>;
 			reg = <0x44220000 0x400>;
 			interrupts-extended =
 				<&intc GIC_SPI 268 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_0 */
